src/common.f90

```diff
@@ -15,6 +15,7 @@ module m_common
       real(dp) :: dx, dy, dz
       integer :: nproc_x = 1, nproc_y = 1, nproc_z = 1
       character(len=20) :: BC_x_s, BC_x_e, BC_y_s, BC_y_e, BC_z_s, BC_z_e
+      logical :: use_fft
```

Comment:
If we plan to have more than just two Poisson solvers, should this be an option instead? e.g. 

```fortran
integer, parameter :: POISSON_SOLVER_FFT = 0, POISSON_SOLVER_CG = 1, POISSON_SOLVER_PCG = 2, ...
...
integer :: poisson_solver_type = POISSON_SOLVER_FFT
select case (poisson_solver_type)
...
```

In fact, might be a good place for an enum (f2003 wow how modern):

```fortran
enum, bind(c)
   enumerator POISSON_SOLVER_FFT, POISSON_SOLVER_CG, POISSON_SOLVER_PCG
endenum
```

Reply:
alternatively (and especially if we need setup) we could do this through the types, e.g.
```
class(poisson_solver), allocatable :: poiss

! This should be an initialisation subroutine...
allocate(poisson_fft :: poiss) ! poisson_cg, etc.
poiss%init() ! Do solver-specific initialisation

call solve(poiss) ! This will type-resolve to the specific Poisson solver
```

Reply:
I see you've mentioned this possibility https://github.com/xcompact3d/x3d2/pull/32#issuecomment-1935747550

Reply:
@semi-h Thoughts on this?

Reply:
I think both your and Paul's suggestions are good, I think I'll leave this as is so that Paul can go ahead with one of the approaches based on how he wants to implement the iterative solver. Is that okay, @pbartholomew08?

Reply:
All done.